Frameworks/Foundation/NSOperation.mm

```diff
@@ -349,4 +355,37 @@ - (void)dealloc {
     [super dealloc];
 }
 
+/**
+ @Status Interoperable
+*/
+- (NSOperationQueuePriority)queuePriority {
+    return _queuePriority;
+}
+
+/**
+ @Status Interoperable
+*/
+- (void)setQueuePriority:(NSOperationQueuePriority)queuePriority {
+    // queuePriority floors to the closest constant to Normal
+    if (queuePriority >= NSOperationQueuePriorityVeryHigh) {
+        _queuePriority = NSOperationQueuePriorityVeryHigh;
+
+    } else if (queuePriority >= NSOperationQueuePriorityHigh) {
+        _queuePriority = NSOperationQueuePriorityHigh;
+
+    } else if (queuePriority > NSOperationQueuePriorityLow) {
+        _queuePriority = NSOperationQueuePriorityNormal;
+
+    } else if (queuePriority > NSOperationQueuePriorityVeryLow) {
+        _queuePriority = NSOperationQueuePriorityLow;
+
+    } else {
+        _queuePriority = NSOperationQueuePriorityVeryLow;
+    }
+}
+
+- (BOOL)_markInQueue {
+    return !_inQueue.fetch_or(YES); // only returns true for the first caller
```

Comment:
nit: since it's truly a `bool`, you'll want `true`; though we're using an `atomic_bool` to back a `BOOL` property o_O